server/service/src/processors/general_processor.rs

```diff
@@ -114,11 +114,12 @@ pub(super) trait Processor {
     fn try_process_record_common(
         &self,
         ctx: &ServiceContext,
+        service_provider: &ServiceProvider,
         changelog: &ChangelogRow,
     ) -> Result<Option<String>, ProcessorError> {
         let result = ctx
             .connection
-            .transaction_sync(|connection| self.try_process_record(connection, changelog))
+            .transaction_sync(|_| self.try_process_record(ctx, service_provider, changelog))
```

Comment:
Transaction within ctx is still in sync, probably need ctx.transaction method ? Anyways need ctx so we can get access to plugin cache